pkg/streaming/commands/processor_create.go

```diff
@@ -167,26 +165,23 @@ func NewProcessorCreateCommand(ctx context.Context, c *cli.Config) *cobra.Comman
 	return cmd
 }
 
-// Parse parameter-stream name bindings, returns parameter and stream names
+// Parse stream bindings, returns potential alias and stream names.
 //
 // Valid values are:
-//  - ${PARAM_NAME}:${STREAM_NAME}
+//  - ${ALIAS}:${STREAM_NAME}
 //  - ${STREAM_NAME}
 //
-// Default values are handled on the server side
-func parseParameterBindings(parameterBindings []string) ([]string, []string) {
-	separator := ":"
-	streamNames := make([]string, len(parameterBindings))
-	parameterNames := make([]string, len(parameterBindings))
-	for i, parameterBinding := range parameterBindings {
-		bindings := strings.SplitAfterN(parameterBinding, separator, 2)
-		if len(bindings) == 2 {
-			parameterName := bindings[0]
-			parameterNames[i] = parameterName[:len(parameterName)-len(separator)]
-			streamNames[i] = bindings[1]
+// Default values are handled on the server side.
+func parseStreamBindings(raw []string) []streamv1alpha1.StreamBinding {
+	bindings := make([]streamv1alpha1.StreamBinding, len(raw))
+	for i, s := range raw {
+		parts := strings.Split(s, ":")
```

Comment:
`SplitAfterN(s, ":", 2)` returns a slice with length <= 2
`Split(s, ":")` returns a slice with length >= 0
if s is `"foo:bar:qix"` e.g., you'd match only `"foo"` as the stream name, which seems incorrect.

Reply:
Neither stream names nor aliases have any reason to contain colons

Reply:
I agree, but I think the CLI should either:
 - reject the alias/stream name
 - send the values as they are (i.e. and not truncate as it is the case now)

Reply:
A steam name with a colon will never resolve, but why is it invalid for an alias? These restrictions should be documented and enforced in the validator.

Reply:
Agreed. There is no shortage of things that we can validate upon, and can be the subject of another PR. For example, aliases should be unique, etc